istio-control/istio-autoinject/templates/mutatingwebhook.yaml

```diff
@@ -29,7 +29,9 @@ webhooks:
         resources: ["pods"]
     failurePolicy: Fail
     namespaceSelector:
-{{- if .Values.sidecarInjectorWebhook.enableNamespacesByDefault }}
+{{- if .Values.sidecarInjectorWebhook.namespaceSelector }}
```

Comment:
I've not had a huge amount of exposure to kustomize we mostly just work with helm charts and try and add the functionality we see a need for as opposed to patching it in. I'm not very familiar with the new installer project as I've not yet ported over to it but is kustomize already used within this project? Just thinking that adding in another tool dependency would cognitively increase the barrier for entry from a usability point of view.

I think the whole `enableNamespacesByDefault` is a little troubling generally since I don't personally find the concept particularly clear myself. I chose to leave it in simply to avoid a breaking change; principally for me I think it'd be ideal if the selector was just fully in configuration with a sensible default allowing users to really drill down what they want but again I chose not to do that to avoid the breaking change.

Right now this chart is largely revolving around the sidecar injector webhook at the minute so I don't see the value in separating it even further; I think that will just make this even harder to understand certainly for newcomers coming into what is already quite a difficult project to get your head around.

The ideal for me would be to scrap everything there currently in regards to the selector and just make it fully configured at a values level; similar to how resources are managed inside a lot of charts.